src/pages/Admin/Charity/Banking/Banking.tsx

```diff
@@ -0,0 +1,67 @@
+import LoaderRing from "components/LoaderRing";
+import { isEmpty } from "helpers";
+import { EMAIL_SUPPORT } from "constants/env";
+import CurrencySelector, { Currency } from "./CurrencySelector";
+import RecipientDetails from "./RecipientDetails";
+import useBanking from "./useBanking";
+
+export default function Banking() {
+  const {
+    currencies,
+    debouncedExpectedFunds,
+    isError,
+    isLoading,
+    targetCurrency,
+    setExpectedFunds,
+    setTargetCurrency,
+  } = useBanking();
+
+  if (isLoading) {
+    return (
+      <div className="flex gap-2">
+        <LoaderRing thickness={10} classes="w-6" /> Loading...
+      </div>
+    );
+  }
+
+  if (isError || !targetCurrency || isEmpty(currencies)) {
+    return (
+      <span>
+        An error occurred. Please try again later. If the error persists, please
+        contact {EMAIL_SUPPORT}.
+      </span>
+    );
+  }
+
+  return (
+    <div className="grid gap-5 justify-start">
+      <CurrencySelector
+        value={targetCurrency}
+        currencies={currencies}
+        onChange={(currency: Currency) => setTargetCurrency(currency)}
+      />
+
+      <div className="flex flex-col gap-2">
+        <label htmlFor="amount">
+          What is amount of donations you expect to keep on our platform?
+        </label>
+        <input
+          id="amount"
+          type="number"
+          onChange={(event) => setExpectedFunds(Number(event.target.value))}
+          className="field-input"
+        />
+      </div>
+
+      {!!targetCurrency && !!debouncedExpectedFunds && (
```

Comment:
Debouncing the input value (see [useBanking.ts](https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2439/files#diff-3124142d81fc04300e4adeef06cfd1a62dc84e080341f3dc827489a150e74417R15)) and loading the form afterwards was a creative choice, so that there's no need for an additional button in the UI (and an additional mouse click for the user).
It is logical to expect the user to input this value once or twice, so the UX should be ok.
What do you think @SovereignAndrey @ap-justin?

Reply:
> It is logical to expect the user to input this value once or twice, so the UX should be ok.
with this, it's even better to make transferring funds multi-step (with buttons `[continue] | [back]`)
1. user sees their balance in USD (e.g. 100USD)
2. decided to withdraw 50USD --> PHP, click `[continue]` and go to next tab/page
`( create quote, recipient requirements )` only fired once

3. user chooses transfer method 
4. user provides details for the chosen transfer method
5. show user summary (fees, exchange rates, expected amount to receive) before they commit to the transfer

Reply:
@ap-justin 
This Banking page was created with the assumption that it would be used as a standalone page, not as part of other flows. So my question about not needing a button here was asked with this in mind.

@SovereignAndrey is Banking page meant to be a part of the withdrawal flow? Didn't see that in the requirements

Reply:
@misicnenad, @SovereignAndrey 

> This Banking page was created with the assumption that it would be used as a standalone page, not as part of other flows.

I'm not sure what this PR ultimately is trying to accomplish, 
is it just trying to save bank details so user would be able to use this information on any operation later on? (hence, the general banking term )

looking at the wise resources used in this PR  -these resources are used for  [transferring money](https://docs.wise.com/api-docs/guides/send-money) (which in our context is like charity withdrawing funds from AP bank )
1. AP profile (source of fund) creates quote 
2. get recipient details based on quote
3. create transfer based on quote and provided recipient details
4. fund transter
5. done 

I am not sure whether recipient created on `2` can be reused on another `quote`(new transfer/different amount). Or if you create a recipient without first creating a quote - would this recipient be valid to be used once you finally create a quote for a later transter ? Also user might opt to receive funds via bank but later on want to receive funds via e-wallet - this is not possible if we only attach a single `recipient_id` on charity's profile

Reply:
> I'm not sure what this PR ultimately is trying to accomplish,

The purpose of this banking page as far as I understand for now (@SovereignAndrey please jump in if I miss anything) is to allow endowments to add their bank details that we can use to:
1. Create a recipient in our Wise account (see https://docs.wise.com/api-docs/api-reference/recipient#create)
2. Send funds to these recipients when relevant endowments request a withdrawal

Regarding 1), you can check the requirements by looking at the big parent Linear issue and its subtasks:
[AP-828: Collect / Setup an Endowments Banking details](https://linear.app/angel-protocol/issue/AP-828/collect-setup-an-endowments-banking-details)

Regarding 2), these are Andrey's words:
![image](https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/19427053/c7acf5a1-d0a2-490d-8332-7ffa336bec06)

**For more context on the types used here and the logic itself, see comments under the related ticket linked below - it explains the types used, the expected format, the reasons for using AWS as proxy to Wise, the reason for dynamic fields etc.:**
[AP-829: Define minimum Wise API fields and shape for creating a recipient `Recipient`](https://linear.app/angel-protocol/issue/AP-829/define-minimum-wise-api-fields-and-shape-for-creating-a-recipient)

**P.S.** I updated this PR's ticket to include the decision to add the "What is amount of donations you expect to keep on our platform" input, see [this comment](https://linear.app/angel-protocol/issue/AP-834/admin-banking-page-posts-bank-details-to-aws-endpoint-and-can-show#comment-ce8b6194).

Reply:
> The purpose of this banking page as far as I understand for now (@SovereignAndrey please jump in if I miss anything) is to allow endowments to add their bank details that we can use to:

Create a recipient in our Wise account (see https://docs.wise.com/api-docs/api-reference/recipient#create)

> Send funds to these recipients when relevant endowments request a withdrawal

> For more context on the types used here and the logic itself, see comments under the related ticket linked below - it explains the types used, the expected format, the reasons for using AWS as proxy to Wise, the reason for dynamic fields etc.:
AP-829: Define minimum Wise API fields and shape for creating a recipient Recipient

@misicnenad 
I am well aware of these, we need recipient to send money. I am not asking what it is needed to created one but rather why we create `recipient` as a standalone entity
by getting requirements from (without quote context)
`GET /v1/account-requirements?source=EUR&target=USD&sourceAmount=1000`

 which we will save for later use, rather than `recipient` being just ephemeral (the only time they provide these details when they withdraw, just like we repeatedly provide our credit card details when purchasing online) and part of a send money process

I believe the reason is stated here, @SovereignAndrey 
<img width="1393" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/0b7a32f8-ac1c-4127-9158-4962fcac2c85">
which says user's wouldn't need to initiate send money process themselves

but going for this is approach has some [caveat](https://docs.wise.com/api-docs/guides/send-money/recipient#create-recipient)
> These factors additionally change on a regular basis, meaning creating and maintaining recipient account can be difficult.

perhaps its safer to just have the user input their bank details? (which is commonly account number, account name) whenever they withdraw.

Reply:
@ap-justin this concern should be directed to @SovereignAndrey

Reply:
resolving this now as got clarifiation from sprint meeting